core/trino-main/src/main/java/io/trino/operator/function/TableFunctionOperator.java

```diff
@@ -285,21 +294,36 @@ public TableFunctionOperator(
         PagesIndex pagesIndex = pagesIndexFactory.newPagesIndex(sourceTypes, expectedPositions);
         HashStrategies hashStrategies = new HashStrategies(pagesIndex, partitionChannels, prePartitionedChannels, sortChannels, sortOrders, preSortedPrefix);
 
-        this.outputPages = pageBuffer.pages()
-                .transform(new PartitionAndSort(pagesIndex, hashStrategies, processEmptyInput))
-                .flatMap(groupPagesIndex -> pagesIndexToTableFunctionPartitions(
-                        groupPagesIndex,
-                        hashStrategies,
-                        tableFunctionProvider,
-                        session,
-                        functionHandle,
-                        properChannelsCount,
-                        passThroughSourcesCount,
-                        requiredChannels,
-                        markerChannels,
-                        passThroughSpecifications,
-                        processEmptyInput))
-                .flatMap(TableFunctionPartition::toOutputPages);
+        if (preferStreaming && passThroughSpecifications.isEmpty() && sortOrders.isEmpty()) {
+            StreamingWorkProcessor streamingWorkProcessor = new StreamingWorkProcessor(processEmptyInput, getOnlyElement(requiredChannels));
```

Comment:
Why `getOnlyElement(requiredChannels)`? We shouldn't make an assumption that there is only one table argument to the table function. Generally, `requiredChannels` is used to split the input into pages corresponding to different table arguments.

Reply:
I have removed this in - https://github.com/trinodb/trino/pull/25046

Reply:
Thanks for the pointers - Figured it out